code/game/objects/items/toys.dm

```diff
@@ -594,8 +594,17 @@
 /obj/random/plushie/item_to_spawn()
 	return pick(subtypesof(/obj/item/toy/plushie) - typesof(/obj/item/toy/plushie/fluff) - typesof(/obj/item/toy/plushie/carpplushie)) //exclude the base type.
 
-/obj/random/plushie/explosive/item_to_spawn()
-	return pick(subtypesof(/obj/item/toy/plushie) - typesof(/obj/item/toy/plushie/fluff) - typesof(/obj/item/toy/plushie/carpplushie))
+/obj/random/plushie/explosive/item_to_spawn() // To override the non explosive random plushie spawn
+	return
+
+/obj/random/plushie/explosive/spawn_item()
+	var/build_path = pick(subtypesof(/obj/item/toy/plushie) - typesof(/obj/item/toy/plushie/fluff) - typesof(/obj/item/toy/plushie/carpplushie))
+	var/obj/item/toy/plushie/plushie = new build_path(src.loc)
+	var/obj/item/I = new /obj/item/grenade/syndieminibomb
+	plushie.has_stuffing = FALSE
+	plushie.grenade = I
+	I.forceMove(plushie)
+	return plushie
```

Comment:
I also think it would be cleaner to put the conditional to add a grenade here, and add the new var to the explosive variant of the randomized generator
```suggestion
/obj/random/plushie/explosive
    var/explosive_chance = 1

/obj/random/plushie/explosive/spawn_item()
	var/obj/item/toy/plushie/plushie = ..()
	if(!prob(explosive_chance)
	    return plushie
	var/obj/item/I = new /obj/item/grenade/syndieminibomb
	plushie.has_stuffing = FALSE
	plushie.grenade = I
	I.forceMove(plushie)
	return plushie
```